lib/ansible/module_utils/facts/system/ssh_pub_keys.py

```diff
@@ -31,22 +31,32 @@ class SshPubKeyFactCollector(BaseFactCollector):
 
     def collect(self, module=None, collected_facts=None):
         ssh_pub_key_facts = {}
-        keytypes = ('dsa', 'rsa', 'ecdsa', 'ed25519')
+        algos = ('dsa', 'rsa', 'ecdsa', 'ed25519')
 
         # list of directories to check for ssh keys
         # used in the order listed here, the first one with keys is used
         keydirs = ['/etc/ssh', '/etc/openssh', '/etc']
 
         for keydir in keydirs:
-            for type_ in keytypes:
-                factname = 'ssh_host_key_%s_public' % type_
+            for algo in algos:
+                factname = 'ssh_host_key_%s_public' % algo
                 if factname in ssh_pub_key_facts:
                     # a previous keydir was already successful, stop looking
                     # for keys
                     return ssh_pub_key_facts
-                key_filename = '%s/ssh_host_%s_key.pub' % (keydir, type_)
+                key_filename = '%s/ssh_host_%s_key.pub' % (keydir, algo)
                 keydata = get_file_content(key_filename)
                 if keydata is not None:
                     ssh_pub_key_facts[factname] = keydata.split()[1]
+                    # Not guaranteed to work with protocol 1, which is deprecated
+                    keyfields = keydata.split()
+                    if len(keyfields) == 4:
```

Comment:
This would fail if there were no options and a comment with 1 space, which is a valid host key which would be handled by sshd fine.

However, I went through the man pages and actually read OpenSSH's code, and I think the man pages are at best ambiguous about host key file formats. They sort of follow the authorized key format. However, the code itself actually takes an entirely different path that looks to have been cargo-culted some time ago from the authorized keys format parser. The options field is never checked for in host public keys, and would make zero sense there anyway. So you can safely ignore that possibility, as sshd would explode with it anyway. I'll report that upstream and see if we can't get it fixed in the man pages.